### PR TITLE
Enable signing daily release build with lifetime certificate

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Enable signing of daily builds with a lifetime certificate. This will allow us to release a signed build but with a certificate which has a small expiry.

MSIX does not support the lifetime cert, hence keep it unchanged.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: tools/releaseBuild/azureDevOps/templates/shouldSign.yml [hunk 1]
```diff
@@ -1,9 +1,12 @@
 steps:
 - powershell: |
     $shouldSign = $true
-    if($env:BUILD_REASON -ne 'Manual')
```
Comment (rjmholt) [inline tools/releaseBuild/azureDevOps/templates/shouldSign.yml:4]: Does this mean that builds that are not manual and are not daily will now be signed with the standard Authenticode cert?

Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/templates/shouldSign.yml:4]: I believe we do not have any builds for this pipeline that are not manual and not daily. The builds we have are:

1. Daily build - kicked off automatically so not manual. -> signed 
2. Manual kicked off for release - so not daily
3. Manually kicked off for testing - so not daily

The pipeline also has an option to skip signing using the pipeline schedule time parameter.

Comment (rjmholt) [reply tools/releaseBuild/azureDevOps/templates/shouldSign.yml:4]: No automated on-push branch build?

**Review by rjmholt — APPROVED** [review]

File: tools/releaseBuild/azureDevOps/templates/shouldSign.yml (hunk 1 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/templates/shouldSign.yml:4]: on-push is disabled for the pipeline.

**Pull request merged**